services/command-router-base/src/main/java/org/eclipse/hono/commandrouter/impl/CommandRouterServiceImpl.java

```diff
@@ -176,6 +200,46 @@ private Duration getSanitizedLifespan(final Duration lifespan) {
                 .otherwise(t -> CommandRouterResult.from(ServiceInvocationException.extractStatusCode(t)));
     }
 
+    @Override
+    public Future<CommandRouterResult> enableCommandRouting(final List<String> tenantIds, final Span span) {
+        Objects.requireNonNull(tenantIds);
+        final boolean isProcessingRequired = tenantsToEnable.isEmpty();
+        tenantsToEnable.addAll(tenantIds);
+        if (isProcessingRequired) {
+            processTenantQueue();
+        }
```

Comment:
As the case of an `enableCommandRouting` invocation while `processTenantQueue()` is still in progress is expected here, I think it would be good to skip already handled tenant entries in `processTenantQueue` (e.g. by means of an `alreadyProcessed` Set as `processTenantQueue()` parameter).

Reply:
I do not really see the value of this information. If tenant IDs are being processed, there will be corresponding FOLLOWS_FROM spans. If not, there won't be any.

Reply:
I don't quite follow. To rephrase my comment: When there are for example 10 adapters making `enableCommandRouting` requests almost in parallel with the same 10 tenants, it would be good for the `processTenantQueue()` loop to skip duplicate tenant entries, e.g. by keeping track of already handled tenant ids via a `alreadyProcessed` `Set` passed around as `processTenantQueue()` parameter.